utils/build/docker/php/parametric/server.php

```diff
@@ -177,6 +157,26 @@ function remappedSpanKind($spanKind) {
     $headers = \DDTrace\generate_distributed_tracing_headers();
     return jsonResponse(["http_headers" => array_map(null, array_keys($headers), $headers)]);
 }));
+$router->addRoute('POST', '/trace/span/extract_headers', new ClosureRequestHandler(function (Request $req) use (&$spans, &$spansDistributedTracingHeaders) {
+    $headers = arg($req, 'http_headers');
+    # Hack: Create a span from the extracted headers to retrieve the parent_id in the distrivuted tracing headers
+    $span = \DDTrace\start_span();
+    $headers = array_merge(...array_map(fn($h) => [strtolower($h[0]) => $h[1]], $headers));
+    $callback = function ($headername) use ($headers) {
+        return $headers[$headername] ?? null;
+    };
+    \DDTrace\consume_distributed_tracing_headers($callback);
+    try {
+        $spanID = $span->parentId;
+    }
+    catch(Error $e) {
+        $spanID = null;
+    }
```

Comment:
```suggestion
    $spanID = $span->parentId ?? null;
```
What's that a weird construction? catching the uninitialized state?

Reply:
Yup, if `consume_distributed_tracing_headers` fails to extract span data from headers (say if the headers list is empty or contains invalid data) then accessing `$span->parentId` raises a `Typed property XXX must not be accessed before initialization` error.

Reply:
Yeah, that you do guard against with `??` in PHP.